tests/file_storage/tests.py

```diff
@@ -401,6 +404,47 @@ def test_delete_no_name(self):
         with self.assertRaises(AssertionError):
             self.storage.delete('')
 
+    def test_copy_file(self):
+        f = ContentFile('content content')
+        src_name = self.storage.save('initial.file', f)
+        dst_name = self.storage.get_available_name('destination.file')
+        self.storage.copy(src_name, dst_name)
+        self.assertTrue(self.storage.exists(src_name))
+        self.assertTrue(self.storage.exists(dst_name))
+
+    def test_copy_file_no_suspicious_operation(self):
+        f = ContentFile('content content')
+        src_name = self.storage.save('initial.file', f)
+        with self.assertRaises(SuspiciousOperation):
+            self.storage.copy(src_name, '/tmp/passwd')
+
+    def test_copy_raise_on_overwrite(self):
+        f = ContentFile('content content')
+        existing_name = self.storage.save('existing.file', f)
+        src_name = self.storage.save('src.file', f)
+        with self.assertRaises(ValueError):
```

Comment:
prefer `self.assertRaisesMessage` to check the message too (need to inherit `SimpleTestCase` for that)